activerecord/lib/active_record/associations/association.rb

```diff
@@ -193,9 +193,25 @@ def creation_attributes
           attributes
         end
 
+        def instantiation_attributes(record)
+          attributes = {}
+
+          owners_name = owner.class.base_class.name.downcase
```

Comment:
@matthewd I think assuming that the correct name is returned from the reflection seems quite reasonable, maybe this is one possible solution?

```
association = record.class.reflect_on_all_associations.find { |ref| ref.klass == owner.class }
correct_association_name = association.name
```

But I think it's ugly and hackish